src/test/java/com/xwiki/authentication/saml/XWikiMock.java

```diff
@@ -0,0 +1,132 @@
+package com.xwiki.authentication.saml;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.StringProperty;
+import com.xpn.xwiki.objects.classes.BaseClass;
+import com.xpn.xwiki.objects.classes.StringClass;
+import com.xpn.xwiki.objects.meta.StringMetaClass;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.rendering.syntax.Syntax;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+
+public class XWikiMock extends XWiki {
+    final Map<String, List<BaseObject>> createdObjectsByEntity = new LinkedHashMap<>();
```

Comment:
I noticed the use of `LinkedHashMap` here and elsewhere. If you don't have a specific use case for it, just use `HashMap`.